### ISSUE TITLE
508-defect-1 ❗ Launchblocker: File for Debt resolution, Select payroll deduction, Unlabeled link and other feedback

### ISSUE THREAD

**Issue by AngelaFowler82** [department-of-veterans-affairs/va.gov-team#22888]
Label: 22685

# 508-defect-1 :exclamation: Launchblocker

https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-1

Issue Title: Defect level 1, launch blocker! File for Debt resolution, Select payroll deduction, Unlabeled link and other feedback

Point of Contact: Angela Fowler
**VFS Point of Contact:** _First name only: Angela

## User Story or Problem Statement:
When accessing a form component using screen reading technology, I expect all links to be labeled and to be able to tab to them. I also expect the purpose of form elements to be consistently spoken, and error messages to be easily accessible and relevant.

## Details
When testing the add payroll deduction component under current job, I encountered the following issues:
<ul>
<li>The link to edit a current deduction is not labeled, and is accessible only when the user arrows to it. </li>
<li>The link to add a deduction is accessible only when the user arrows to it.</li>
<li>The first time I access with a given test account, when tabbing from the gross monthly income edit box to the box which asks the user to choose a deduction, the purpose of the box is not read. </li>
<li> When I place focus on a deduction and press "tab," the selection is not chosen. This is a common method by which screen reader users make selections in list boxes. It would be helpful if screen reader users were instructed to press "enter" on their selection. </li>
<li>When I select "Add payroll deduction" after having saved a deduction, JAWS says "Error please provide a response." Since I am opening a new component, there is no opportunity to provide the response before the error message was spoken.</li>
<li>I was able to enter a garnishment which was higher than my stated gross income and proceed to the next section of the form with no error message. </li>
</ul>

## Acceptance Criteria
 - [ ] The link to edit a deduction is labeled and included in the tab order,
 - [ ] The link to add a deduction is included in the tab order,
 - [ ] The purpose of The deduction selection element is stated consistently,
 - [ ] When selecting a deduction, an instruction is given that screen reader users must press "enter" on their selection,
 - [ ] When the user selects "add payroll deduction," they are not given an error message unless they tab through the form without responding to a required question,
 - [ ] The form returns an error if a deduction is higher than the user's stated gross income,

**Comment by denisecoveyduc** [comment]
@joshkimux @AngelaFowler82  I am doing a little GH clean up and wondered if you could just take a quick look at this issue for me to validate that this is still a concern? Since (and before I joined the team) the FSR has gone through many changes and this may or may not have already been addressed. Thanks in advance for checking it out for us.

**Comment by AngelaFowler82** [comment]
At this point, the component to add payrole deductions is still problematic. Once a screen reader enters their gross income, they tab to a combo box which is unlabeled, the screen reader simply reads "Autosuggest input." Beginner screen readers may not know what to do with this. If the user tabbs down, they are presented with a list of possible deductions. Once they choose one, tab to the next edit field and enter the amount, they can tab to a button which says "Save payrole deductions." When they choose that button, focus does not change. There is no indication as to whether their deduction is saved or whether they can add a new deduction or search for a link to do so. Also, the "Add payrole deduction" is not part of the tab order and must be accessed using the arrow keys.

**Comment by denisecoveyduc** [comment]
Shifting this over to our Tech debt epic to ensure we can address this and move it forward. It was originally captured as part of the staging review for the initial FSR launch.

**Comment by joshkimux** [comment]
@denisecoveyduc did this get wrapped into the tech debt epic? And if so, is there a new ticket that could replace it? I could then close this issue out.

**Comment by mgleo** [comment]
https://staging.va.gov/manage-va-debt/request-debt-help-form-5655/deduction-checklist

[Private Zenhub Image](https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBaFhMIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--5d26d1b4956318745e0e0bbbc5b3d6c771633b5d/Screenshot%202023-09-22%20at%2011.30.21%20AM.png)
I believe this has been resolved with the eFSR list loop improvements @HeatherRienks @Tom-Davis

**Comment by Tom-Davis** [comment]
The product has evolved beyond the need for this - closing as no longer needed.
